### PR TITLE
Allow to set dimensions for metrics

### PR DESCRIPTION
**Pull request by sumo-drosiek**
It would be nice to add some dimensions directly from configuration.
`custom_dimensions` allows to add dimensions in `X-Sumo-Dimensions` header.

https://help.sumologic.com/03Send-Data/Sources/02Sources-for-Hosted-Collectors/HTTP-Source/Upload-Metrics-to-an-HTTP-Source#supported-http-headers

### CODE REVIEW

File: lib/fluent/plugin/out_sumologic.rb [hunk 1]
```diff
@@ -37,6 +37,10 @@ def request_headers(source_host, source_category, source_name, data_type, metric
       else
         raise RuntimeError, "Invalid #{metric_data_format}, must be graphite or carbon2 or prometheus"
       end
+
+      unless metadata.nil?
+        headers['X-Sumo-Metadata'] = metadata
```
Comment (frankreno) [inline lib/fluent/plugin/out_sumologic.rb:42]: As discussed on Slack, let's confirm Metadata is the best use, as opposed to dimensions.

File: lib/fluent/plugin/out_sumologic.rb [hunk 2]
```diff
@@ -136,6 +143,10 @@ def configure(conf)
       conf['custom_fields'] = nil
     end
 
+    if conf['custom_metadata'].nil? || conf['custom_metadata'].strip.length == 0
```
Comment (frankreno) [inline lib/fluent/plugin/out_sumologic.rb:146]: Should we do some validation on the configuration? Testing it is actually a KVP string?

Comment (sumo-drosiek) [reply lib/fluent/plugin/out_sumologic.rb]: I added simple validation :+1:

File: lib/fluent/plugin/out_sumologic.rb (hunk 1 above)
Comment (sumo-drosiek) [reply lib/fluent/plugin/out_sumologic.rb]: I change it to dimensions, but regarding another discussion I'm not sure if we want to merge those changes or not

**Comment by sumo-drosiek** [comment]
U updated PR description due to changes

**Pull request merged**